components/clock.rst

```diff
@@ -207,6 +212,48 @@ being in a month or another.
 
     The :class:`Symfony\\Component\\Clock\\ClockAwareTrait` was introduced in Symfony 6.3.
 
+.. _clock_date-point:
+
+The ``DatePoint`` Class
+-----------------------
+
+When using the Clock component, you manipulate instances of
+:class:`Symfony\\Component\\Clock\\DatePoint`. The main purpose of this
+class is to leverage the Clock component to get the "current time". When
+creating a new ``DatePoint`` object, the date and time are fetched
+from the :class:`Symfony\\Component\\Clock\\Clock` class. This means
+that if you did any changes to the clock as stated in the
+:ref:`usage section <clock_usage>`, it will be reflected when creating
+a new ``DatePoint``. This can be particularly useful when
+:ref:`writing time-sensitive tests <clock_writing-tests>`.
+
+Creating a new ``DatePoint`` is as straightforward as creating other date
+objects::
+
+    use Symfony\Component\Clock\DatePoint;
+
+    // create a DatePoint set at the current date and time
+    $now = new DatePoint();
+
+    // you can specify a timezone
+    $withTimezone = new DatePoint(timezone: new \DateTimezone('UTC'));
+
+    // you can also create a DatePoint from a reference date
+    $referenceDate = new \DateTimeImmutable();
+    $relativeDate = new DatePoint('+1month', reference: $referenceDate);
+
+``DatePoint`` extends ``DateTimeImmutable`` so that you can use it seamlessly
+everywhere a ``DateTimeImmutable`` is needed. In addition ``DatePoint`` offers
+stricter return types and provides consistent error handling across versions of
+PHP, thanks to polyfilling `PHP 8.3's behavior`_ on the topic.
```

Comment:
I propose doing a slightly different order, the fact that `DatePoint` is "just" a `DateTimeImmutable` is important for a user, so we should mention it early imho.

Besides that, we can be a bit less verbose. Implementation details are great for PR discussions, usage details are what matters for documentation :)

```suggestion
The Clock component uses a special :class:`Symfony\\Component\\Clock\\DatePoint`
class. This is a small wrapper on top of PHP's :phpclass:`DateTimeImmutable`.
You can use it seemlessly everywhere a :phpclass:`DateTimeImmutable` or
:phpclass:`DateTimeInterface` is expected.

The ``DatePoint`` object fetches the date and time from the
:class:`Symfony\\Component\\Clock\\Clock` class. This means that if you did any
changes to the clock as stated in the :ref:`usage section <clock_usage>`, it
will be reflected when creating a new ``DatePoint``.

You can also create a new ``DatePoint`` instance directly, for instance when
using it as a default value::

    use Symfony\Component\Clock\DatePoint;

    class Post
    {
        public function __construct(
            // ...
            private \DateTimeImmutable $createdAt = new DatePoint()
        ) {
        }
    }

The constructor also allows setting a timezone or custom referenced date:

    // you can specify a timezone
    $withTimezone = new DatePoint(timezone: new \DateTimezone('UTC'));

    // you can also create a DatePoint from a reference date
    $referenceDate = new \DateTimeImmutable();
    $relativeDate = new DatePoint('+1month', reference: $referenceDate);

.. note::

    In addition ``DatePoint`` offers stricter return types and provides consistent
    error handling across versions of PHP, thanks to polyfilling `PHP 8.3's
    behavior`_ on the topic.
```

Reply:
Updated, thank you Wouter!